tests/framework/shim/unix_shim.cpp

```diff
@@ -150,7 +150,15 @@ FRAMEWORK_EXPORT DIR* OPENDIR_FUNC_NAME(const char* path_name) {
 
 FRAMEWORK_EXPORT struct dirent* READDIR_FUNC_NAME(DIR* dir_stream) {
 #if !defined(__APPLE__)
-    if (!real_readdir) real_readdir = (PFN_READDIR)dlsym(RTLD_NEXT, "readdir");
+    if (!real_readdir) {
+        if (sizeof(void*) == 8) {
+            real_readdir = (PFN_READDIR)dlsym(RTLD_NEXT, "readdir");
```

Comment:
FWIW I'm pretty sure `readdir64` also exists on 64-bit, so this could maybe be unconditional?

Reply:
Since this code works on non-gnu platforms (ie, apple) I chose to not rock the boat too hard and only use the readdir64 when I know I need it (32 bit linux). Its the test framework as well, so no hard to fix if needed in the future.

Reply:
Makes sense.